operator/roles/kiali-deploy/tasks/main.yml

```diff
@@ -424,15 +425,44 @@
   - current_image_name is defined and current_image_version is defined
   - (current_image_name != kiali_vars.deployment.image_name) or (current_image_version != kiali_vars.deployment.image_version)
 
+# Get the deployment's custom annotation we set that tells us when we last updated the Deployment.
+# We need this to ensure the Deployment we update retains this same timestamp unless changes are made
+# that requires a pod restart - in which case we update this timestamp.
```

Comment:
I would have thought when we update the deployment we just set the annotation to force the update. Why do we need the previous value?

Reply:
We need to get the previous value because if there IS NO previous value, we have to set it to the current time.

See line 451, "if the current_deployment_last_update" is not defined (meaning there is no deployment right now) then line 448 creates one with the current time and we set `deployment_is_new` to true. But without this attempt to get the current deployment, we have no idea if the deployment is new or if it already exists from a prior reconciliation run.

We need that `deployment_is_new` flag as well as part of the condition to know when we need to update the deployment so it does the rolling update. Line 511 - so if the deployment is new we don't do the task on line 504 ("- name: Force the Kiali pod to restart if necessary") regardless of the fact that the configmap changed because we already know, since its new, the pod is coming up fresh anyway.